lib/index.js

```diff
@@ -0,0 +1,645 @@
+   * Builds a graph using OTP
+   */
+  async buildGraph () {
+    // in build mode, use execa which simplifies a few things
+    // The execa options argument creates a combined stdout and stderr output
+    // stream via setting the `all` flag to true.
+    const args = getBaseOTPArgs(this.manifest.jarFile)
+    args.push('--build')
+    args.push(path.join(this.manifest.graphsFolder, this.manifest.routerName))
+    const subprocess = execa('java', args, { all: true })
+
+    // keep the last 100 logs to stdout/stderr in memory
+    const last100Logs = new CircularBuffer(100)
+
+    // Pipe all output to a logfile
+    subprocess.all.pipe(fs.createWriteStream(this.manifest.buildLogFile))
+    // Analyze the OTP stdout and stderr, storing logs into the CircularBuffer.
+    subprocess.all.on('data', (data) => {
+      const lastMessage = data.toString().trim()
+      if (lastMessage !== '') {
+        last100Logs.push(lastMessage)
+      }
+    })
+
+    // wait 1 second prior to entering loop. This is mainly just to let the
+    // subprocess stream logging catch up when testing.
+    await waitOneSecond()
+
+    // Check on OTP as it starts up. Wait until graph building is complete. Update
+    // the status as graph build progresses.
+    while (subprocess.exitCode === null) {
+      await waitOneSecond()
+      // Update status with the latest message from the OTP logs
+      const lastLog = last100Logs.size() > 1
+        ? (
+          last100Logs
+            .get(last100Logs.size() - 1) // get the most recent entry
+            .replace(/^\d\d:\d\d:.*\(.*\)\s*/, '') // strip java timestamp and class
+            .substring(0, 60)
+        )
+        : ''
+      await this.updateStatus(`Building graph... ${lastLog !== '' ? ` (${lastLog})` : ''}`)
+    }
+
+    // OTP exited, check if it was successful and do something if not
+    if (subprocess.exitCode > 0) {
+      this.log.error(last100Logs.toarray().join('\n'))
+      // immediately upload logs
+      await this.uploadBuildLogs()
+      await this.fail('Build graph failed! Please see logs.')
+    }
+    this.status.graphBuilt = true
+  }
+
+  /**
+   * Starts OTP as a server
+   */
+  async startServer () {
+    const args = getBaseOTPArgs(this.manifest.jarFile)
+    args.push('--server')
+    args.push('--graphs')
+    args.push(this.manifest.graphsFolder)
+    args.push('--router')
+    args.push(this.manifest.routerName)
+
+    // if running as a server, use native child_process library instead of
+    // execa because it does not seem possible to keep writing output to logs
+    // while running detached processes with execa.
+    const out = fs.openSync(this.manifest.serverLogFile, 'w')
+    const otpServerProcess = spawn(
+      'java',
+      args,
+      { detached: true, stdio: [ 'ignore', out, out ] }
+    )
+    otpServerProcess.unref()
+    await this.updateStatus('Starting OTP server...')
+    this.log.info(`java ${args.join(' ')} running as pid ${otpServerProcess.pid}`)
```

Comment:
This prompt should also remind the default port (8080) to the user.

Reply:
That info would be in the server startup logs. Lmk if you feel it's really needed here too.